@kernel macro for creating kernel (#38)
JuliaGaussianProcesses/KernelFunctions.jl: docs/src/userguide.md

To create a kernel chose one of the kernels proposed, see [Kernels](@ref), or create your own, see [Creating Kernels](@ref)
For example to create a square exponential kernel

```julia
  k = SqExponentialKernel()
```

> Instead of having lengthscale(s) for each kernel we use `Transform` objects (see [Transform](@ref)). The transform operations are going to be applied on the inputs before they are passed to the kernel.
> For example, the [ScaleTransform](@ref) multiply every sample by a scalar $\rho$. A `SqExponentialKernel` with a `ScaleTransform(ρ)`, is therefore equivalent to have a `SqExponentialKernel` with lengthscale `1/ρ`.
> Here are some examples on how to use these transformations and are all equivalent:

Reviewer: ```suggestion
Instead of having lengthscale(s) for each kernel we use `Transform` objects (see [`Transform`](@ref)). The transformations are applied to the inputs before the kernel is evaluated.
For example, the [`ScaleTransform`](@ref) multiplies every sample with a scalar. A `SqExponentialKernel` with a `ScaleTransform(ρ)` is therefore equivalent to a `SqExponentialKernel` with lengthscale `1/ρ`.
Here are some examples of how to use these transformations that are all equivalent:
```
